designs/extensible-cli-and-scaffolding-plugins-phase-1.md

```diff
@@ -50,6 +51,38 @@ existing qualifier defined by the plugin. FQ names prevent conflicts between
 plugin names; the plugin runner will ask the user to add a name qualifier to
 a conflicting plugin.
 
+#### Plugin versioning
+
+Plugin versions are valid "tolerant" semver, in that they can optionally contain
+a "v" prefix and patch version is not used nor allowed. Changes made to plugins
+may require a version increase depending on whether they break the current version or not.
+See the [contribution guide][../CONTRIBUTING.md#example] for a version change example.
+
+##### Breaking changes
+
+Situations that result in a breaking change:
+* Deleting a default scaffold file.
+* Modifying a scaffold file that will be modified by other commands.
+* Changing the path to a default scaffold file.
+
+Simply put, any change that will break the expected behavior of a project built
+with the previous _minor_ plugin version is a breaking change.
+
+Breaking changes require an increase in a plugin's _major_ version.
+
+##### Non-breaking changes
+
+There are two types of non-breaking changes: those requiring a _minor_ version
+increase, and those that do not require a version be changed.
+
+Situations that require a _minor_ version bump:
+* Adding a new file that other commands do not assume exists.
+* Modifying a scaffold file in a way that does not affect other commands.
+
+Situations that do not require a version be changed:
+* Typo fixes.
+* Cosmetic changes that do not affect program behavior.
```

Comment:
it is not always true. Typo fixes can be a breaking change. E.g 1498
So, I'd suggest just remove it since you detailed enough on top.
```suggestion
```

Reply:
#1498 is not a typo fix, it changes scaffolding.

Reply:
If the typo fix is not in the scaffold files then we will not change the plugin at all.